xbmc/utils/BufferObjectFactory.cpp

```diff
@@ -8,17 +8,22 @@
 
 #include "BufferObjectFactory.h"
 
-std::vector<std::function<std::unique_ptr<CBufferObject>()>> CBufferObjectFactory::m_bufferObjects;
+std::list<std::function<std::unique_ptr<CBufferObject>()>> CBufferObjectFactory::m_bufferObjects;
 
 std::unique_ptr<CBufferObject> CBufferObjectFactory::CreateBufferObject()
 {
-  return m_bufferObjects.back()();
+  for (const auto bufferObject : m_bufferObjects)
+  {
+    return bufferObject();
```

Comment:
Maybe that's a stupid question and not related to this PR, but why can you have multiple factories when only the last registered one ever gets used anyway?

Reply:
This makes sense to me, you use a stack to give priority to the most recent factory registered. Then in your registration class you pessimistically test for the existence of "worse" buffers before "better" ones.

Reply:
Actually, the comment clears it up: "In the future this may include some criteria for selecting a specific CBufferObject derived type. Currently it returns the CBufferObject implementation that was registered last."

@lrusak With your patch though, it returns the one registered *first*, am I correct?

Reply:
@yol I don't think so, he's using `m_bufferObjects.emplace_front(createFunc);` now and returns the first object he finds which is still the last that was registered.

Reply:
Makes sense!